hackathon_site/dashboard/frontend/src/components/teamDetail/TeamInfoTable/TeamInfoTable.test.tsx

```diff
@@ -1,41 +1,66 @@
-import { render, screen } from "testing/utils";
-import { mockTeamMultiple } from "testing/mockData";
+import { makeMockApiResponse, render, waitFor, when } from "testing/utils";
+import { mockTeam } from "testing/mockData";
+import { get } from "api/api";
 
 import TeamInfoTable from "components/teamDetail/TeamInfoTable/TeamInfoTable";
+import { makeStore } from "slices/store";
 
-describe("team info table", () => {
-    test("renders team info table", () => {
-        const { container } = render(<TeamInfoTable />);
-        const checkboxes = container.getElementsByClassName("MuiCheckbox-root");
-
-        expect(screen.getByText("Team info")).toBeInTheDocument();
-
-        for (let i = 0; i < mockTeamMultiple.profiles.length; i++) {
-            // renders all user names
-            expect(
-                screen.getByText(
-                    `${mockTeamMultiple.profiles[i].user.first_name} ${mockTeamMultiple.profiles[i].user.last_name}`
-                )
-            ).toBeInTheDocument();
-
-            // renders all user emails
-            expect(
-                screen.getByText(mockTeamMultiple.profiles[i].user.email)
-            ).toBeInTheDocument();
-
-            // renders all user phone numbers
-            expect(
-                screen.getByText(mockTeamMultiple.profiles[i].user.phone)
-            ).toBeInTheDocument();
-
-            // renders checkboxes correctly checked
-            // NOTE: this test assumes that users are rendered in the table in the same order that they are stored in the json test data
-            if (mockTeamMultiple.profiles[i].id_provided) {
-                expect(checkboxes[i].classList.contains("Mui-checked")).toBe(true);
-            } else {
-                expect(checkboxes[i].classList.contains("Mui-checked")).toBe(false);
-            }
-        }
+jest.mock("api/api", () => ({
+    ...jest.requireActual("api/api"),
+    get: jest.fn(),
+}));
+
+const mockedGet = get as jest.MockedFunction<typeof get>;
+
+describe("Team info table", () => {
+    it("Renders team info table", async () => {
+        const teamInfoApiResponse = makeMockApiResponse(mockTeam);
+
+        when(mockedGet)
+            .calledWith("/api/event/teams/1/")
+            .mockResolvedValue(teamInfoApiResponse);
+
+        const { getByText, queryByTestId, getAllByRole } = render(<TeamInfoTable />);
+
+        const firstNames = mockTeam.profiles.map((user) => {
+            return user.user.first_name;
+        });
+        const lastNames = mockTeam.profiles.map((user) => {
+            return user.user.last_name;
+        });
+        const emails = mockTeam.profiles.map((user) => {
+            return user.user.email;
+        });
+        let numIdsProvided = 0;
+        mockTeam.profiles.forEach((user) => {
+            if (user.id_provided) numIdsProvided++;
+        });
+
+        await waitFor(() => {
+            expect(queryByTestId("team-info-linear-progress")).toBeInTheDocument();
```

Comment:
your tests are failing because of this assertion. You don't call the api in this component so the loading spinner and the store will never be populated. 
You should split up this test: populate the store in the table test (you can use mockStoreWithEntities function from utils.ts) and verify the content here, then in TeamDetail.test.tsx mock the API and verify that it's being called when the page renders.